sg_jira/handlers/entity_issue_handler.py

```diff
@@ -987,18 +975,114 @@ def _get_shotgun_assignment_from_jira_issue_change(
+        :raises InvalidJiraValue: Raised if the user could not be found and ``raise_on_missing_user`` is True.
+        """
+        if jira_user is not None:
+            emailAddress = jira_user["emailAddress"]
+        elif user_id is not None:
+            emailAddress = self._jira.user(user_id).emailAddress
+        else:
+            raise RuntimeError("jira_user or user_id cannot be both None.")
+
+        sg_user = self._shotgun.find_one(
+            "HumanUser",
+            [["email", "is", emailAddress]],
+            ["email", "name"]
+        )
+        if not sg_user:
+            if raise_on_missing_user:
+                raise InvalidJiraValue(
+                    shotgun_field,
+                    jira_user,
+                    "Unable to find a Shotgun user with email address %s" % (
+                        emailAddress
+                    )
                 )
-                if not sg_user:
+            else:
+                self._logger.debug(
+                    "Unable to find a Shotgun user with email address %s" % (
+                        jira_user.emailAddress
+                    )
+                )
+        return sg_user
+
+    def _jira_cloud_user_to_shotgun(self, shotgun_field, user_id, jira_user=None, raise_on_missing_user=True):
+        """
+        Resolve the Shotgun user associated to the JIRA user passed in.
+
+        This method should be called against a JIRA Cloud server.
+
+        :param str shotgun_field: Field to sync the value to in Shotgun.
+        :param str user_id: Value of the to or from of a JIRA changelog.
+        :param dict jira_user: User resource, typically the assignee field on an issue. Can be None
+        :param bool raise_on_missing_user: Indicate how to handle unknown users.
+
+        :returns: A Shotgun user entity dictionary.
+        :raises InvalidJiraValue: Raised if the user could not be found and ``raise_on_missing_user`` is True.
+        """
+        # If the jira_user has been passed in, just use the accountId!
+        if jira_user is not None:
+            user_id = jira_user["accountId"]
+        # We didn't receive the jira_user, this happens when the from field is
+        # passed in to this method, so we'll have use the user id passed in.
+        #
+        # Interestingly, when the a user field is updated via the JIRA API,
+        # the username is passed in instead of the account id, so we'll have to
+        # resolve it.
+        elif self.ACCOUNT_ID_RE.match(user_id) is None:
+            self._logger.debug("The changelog's to/from contains a user name. accountId will be retrieved.")
+            user = self._jira.user(user_id)
+            if not user:
+                if raise_on_missing_user:
                     raise InvalidJiraValue(
                         shotgun_field,
                         jira_user,
-                        "Unable to find a Shotgun user with email address %s" % (
```

Comment:
This is code from the original user matching in the original method. Github's diff is making out lives harder here.